### DISCUSSION TITLE
OperatingSystem.Current

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#45228] [Q&A] [answered]
Currently the ```OperatingSystem``` class allows us to check if the operating system is a specific platform/version.

It would be nice if it also exposed a ```Current``` property that had the following members:
```
public abstract class OperatingSystem {
  public string Name {get; }
  public Version Version {get; }
}

public class WindowsOperatingSystem : OperatingSystem {
  public string ServicePack {get; }
}

public class LinuxOperatingSystem : OperatingSystem {
  public string Distribution {get; }
}
//etc
```

This would provide an easy way to determine the current operating system and detect relevant information about it.

**Comment by huoyaoyuan** [comment, marked as answer]
It should be intentionally to NOT provide such information in core api.

Versioning schema for future OS are effectively unknown. For example, service pack was a thing until Win7, but it isn't a thing in Win10.
Another reason is that it's hard to take correct decision for future OS. The suggested approach is always checking "is version at least" or "is feature available".

And also, taking versioning decision based on string is super stupid. There are rumor said that `osVersion.StartWith("Windows 9")` for detecting win95-98 forces a jump from Windows 8 to Windows 10.
